### ISSUE TITLE
tabs-motion does not support extending `<Tab />`

### ISSUE THREAD

**Issue by shhac** [uber/baseweb#5250]
In `tabs-motion`, [`Tab` is simply a `() => null`](https://github.com/uber/baseweb/blob/master/src/tabs-motion/tab.ts) and all functionality is handled in [`Tabs`](https://github.com/uber/baseweb/blob/master/src/tabs-motion/tabs.tsx) by iterating over `props.children` and rendering in other components over which the user has no control.

https://github.com/uber/baseweb/blob/7f2d89bce6548366966cbaa48f9bebf2516a2c2f/src/tabs-motion/tabs.tsx#L255-L259

This is great in the sense that we get the headings and render-on active for "free", but it means some simple code such as

```ts
const MyTab = (): JSX.Element => {
  return <Tab
    key="tab-1"
    overrides={{
      Tab: { style: () => ({ outline: '1px solid red'; }) }
    }}
  >Hello world</Tab>
};
```

If we prop-match `MyTab`'s signature to `TabProps` then we get the behaviour equivalent to `Tab`'s, but we can't extend or change behaviour as the function body is effectively never run

```ts
// i.e. this won't work

interface MyTabProps extends TabProps {
  showOutline?: boolean;
}

const MyTab = ({
  showOutline = false,
  ...props
}: MyTabProps) => {
  const overrides = showOutline ? { Tab: { style: () => ({ outline: '1px solid red' }) } } : undefined;
  return <Tab overrides={overrides} {...props} />
};
```

It would be great if this was supported!

---

In the meantime, I've worked around this by adding my own `MyTabs` too which does a `const mappedChildren = Children.map(children, doHardwork)`, making `mappedChildren` look like the children `Tabs` would normally expect -- which feels like propagating an anti-pattern and less than ideal

**Comment by theodoreed** [comment]
I agree, trying to use the TabContent override with Tabs Motion and it does not work. Hopefully this gets fixed in the next update.
